docs/language/references.mdx

```diff
@@ -151,3 +151,39 @@ counterRef3.count  // is `44`
 
 References are ephemeral, i.e. they cannot be [stored](accounts#account-storage).
 Instead, consider [storing a capability and borrowing it](capability-based-access-control) when needed.
+
+### Reference validity
+
+Ephemeral references stay valid throughout the course of the program.
+However, **references to resources** can become invalid during a program if the underlying resource is moved
+or destroyed after taking the reference.
+
+```cadence
+let r <-create R()
+
+// Take a reference to resource.
+let ref = &r as &R
+
+// Then transfer the resource into an account.
+// This will invalidate all the references taken to the resource `r`.
+account.save(<-r, to: /storage/r)
+
+// Static error, since the referenced resource has been moved.
+ref.id = 2
+```
+
+A reference is invalidated upon the first transfer of the underlying resource, regardless of the origin and the destination.
+
+```cadence
+let ref = &r as &R
+
+// Moving the resource to a different vairable would also invalidate the reference.
```

Comment:
```suggestion
// Moving a resource to a different variable invalidates all references of it.
```

Reply:
I think usually we say "references to a value" rather than "references of a value", so maybe "invalidates all references to it" instead?

Reply:
```suggestion
// Moving a resource to a different variable invalidates all references to it.
```